third_party/amd/backend/compiler.py

```diff
@@ -150,7 +160,9 @@ def parse_attr(desc):
     @staticmethod
     def get_arg_specialization(arg, ty, **kwargs):
         ret = BaseBackend.get_arg_specialization(arg, ty, **kwargs)
-        if ty == "tensor" and HIPBackend.is_within_2gb(arg):
+        # Only attempt to do buffer ops specialization if buffer ops are enabled.
+        # Otherwise the is_within_2gb check is unnecessary overhead.
+        if use_buffer_ops() and ty == "tensor" and HIPBackend.is_within_2gb(arg):
```

Comment:
If the string manipulation for cache keys is an overhead, we were also wondering how much it would save for example going to using C++ or another mechanism. Just some quick food for thoughts; haven't worked the details yet.